cloud/endagaweb/views/dashboard.py

```diff
@@ -885,28 +908,404 @@ def _search_events(self, profile, query_string, orig_events):
-
-                # Get any numbers that match, and add their associated
-                # subscribers' events to the results
-                potential_subs = (
-                    Number.objects.filter(number__icontains=query)
-                                  .values('subscriber')
-                                  .filter(subscriber__network=network)
-                                  .distinct())
-                if potential_subs:
-                    events |= (UsageEvent.objects
-                                .filter(subscriber__in=potential_subs))
-
-                res_events |= events
-            return res_events
+        network = profile.network
+        queries = query_string.split()
+
+        res_events = UsageEvent.objects.none()
+        for query in queries:
+            events = orig_events
+            events = (events.filter(kind__icontains=query)
+                      | events.filter(reason__icontains=query)
+                      | events.filter(subscriber__name__icontains=query)
+                      | events.filter(subscriber__imsi__icontains=query)
+                      | events.filter(subscriber_imsi__icontains=query))
+
+            # Get any numbers that match, and add their associated
+            # subscribers' events to the results
+            potential_subs = (
+                Number.objects.filter(number__icontains=query)
+                              .values('subscriber')
+                              .filter(subscriber__network=network)
+                              .distinct())
+            if potential_subs:
+                events |= (UsageEvent.objects
+                           .filter(subscriber__in=potential_subs))
+
+            res_events |= events
+        return res_events
+
+
+class UserManagement(ProtectedView):
+    def get(self, request, *args, **kwargs):
+        # Handles request from Network Admin or Cloud Admin
+        user_profile = UserProfile.objects.get(user=request.user)
+        user = User.objects.get(id=user_profile.user_id)
+        permission_set = ["credit", "graph", "report", "smsbroadcast", "tower",
+                          "bts", "subscriber", "network",
+                          "notification", "usageevent"]
+
+        # View network is restricted
+        #  else giving this in permission will allow user to have all networks
+        restricted_perms = ['view_network']
+
+        if user.is_superuser:  # Cloud Admin
+            role = USER_ROLES
+        else:  # Network Admin
+            role = USER_ROLES[0:len(USER_ROLES) - 1]
+            restricted_perms.extend(
+                ['add_bts', 'change_bts', 'deregister_bts', 'change_network',
+                 'download_report',
+                 'download_graph', 'deactive_subscriber'])
+
+        # Set the context with various stats.
+        content_type = ContentType.objects.filter(app_label='endagaweb',
+                                                  model__in=
```

Comment:
Fixed.  Again it was done due to pylint configuration but for future will keep proper formatting.